.github/workflows/main.yml

```diff
@@ -49,76 +119,150 @@ jobs:
           perl-version: ${{ matrix.perl }} # optional, default is 5
           # The distribution of Perl binary.
 
+      - name: Cache cpan modules
+        id: cache-cpan-modules
+        uses: actions/cache@v2
+        with:
+          # A list of files, directories, and wildcard patterns to cache and restore
+          path: /opt/hostedtoolcache/perl
+          # An explicit key for restoring and saving the cache
+          key: cpan-${{ matrix.perl }}-${{ hashFiles('cpanfile') }}
+
+      - name: Install required features
+        run: |
+          cpm install -g -v --resolver=metacpan --notest \
```

Comment:
I found `cpm` to be much faster than `cpanm` however, the latter allows doesn't install dists which have a satisfied version already installed. `cpm` simply installs all packages named in the list of dependencies, no exceptions.